examples/conway/main.rs

```diff
@@ -0,0 +1,394 @@
+use pixels::{Error, Pixels, SurfaceTexture};
+use winit::dpi::{LogicalPosition, LogicalSize, PhysicalSize};
+use winit::event::{Event, VirtualKeyCode, WindowEvent};
+use winit::event_loop::{ControlFlow, EventLoop};
+use winit_input_helper::WinitInputHelper;
+
+const SCREEN_WIDTH: u32 = 400;
+const SCREEN_HEIGHT: u32 = 300;
+
+fn main() -> Result<(), Error> {
+    env_logger::init();
+    let event_loop = EventLoop::new();
+    let mut input = WinitInputHelper::new();
+    let (window, surface, mut p_width, mut p_height, mut hidpi_factor) =
+        create_window("Conway's Game of Life", &event_loop);
+
+    let surface_texture = SurfaceTexture::new(p_width, p_height, surface);
+
+    let mut life = ConwayGrid::new_random(SCREEN_WIDTH as usize, SCREEN_HEIGHT as usize);
+    let mut pixels = Pixels::new(SCREEN_WIDTH, SCREEN_HEIGHT, surface_texture)?;
+    let mut paused = false;
+
+    let mut draw_state: Option<bool> = None;
+
+    event_loop.run(move |event, _, control_flow| {
+        // The one and only event that winit_input_helper doesn't have for us...
+        if let Event::WindowEvent {
+            event: WindowEvent::RedrawRequested,
+            ..
+        } = event
+        {
+            life.draw(pixels.get_frame());
+            pixels.render();
+        }
+
+        // For everything else, for let winit_input_helper collect events to build its state.
+        // It returns `true` when it is time to update our game state and request a redraw.
+        if input.update(event) {
+            // Close events
+            if input.key_pressed(VirtualKeyCode::Escape) || input.quit() {
+                *control_flow = ControlFlow::Exit;
+                return;
+            }
+            if input.key_pressed(VirtualKeyCode::P) {
+                paused = !paused;
+            }
+            if input.key_pressed(VirtualKeyCode::Space) {
+                // Space is frame-step, so ensure we're paused
+                paused = true;
+            }
+            if input.key_pressed(VirtualKeyCode::R) {
+                life.randomize();
+            }
+            // Handle mouse. This is a bit involved since support some simple
+            // line drawing (mostly because it makes nice looking patterns).
+            let (mouse_cell, mouse_prev_cell) = input
+                .mouse()
+                .map(|(mx, my)| {
+                    let (dx, dy) = input.mouse_diff();
+                    let prev_x = mx - dx;
+                    let prev_y = my - dy;
+                    let dpx = hidpi_factor as f32;
+                    let (w, h) = (p_width as f32 / dpx, p_height as f32 / dpx);
+                    let mx_i = ((mx / w) * (SCREEN_WIDTH as f32)).round() as isize;
+                    let my_i = ((my / h) * (SCREEN_HEIGHT as f32)).round() as isize;
+                    let px_i = ((prev_x / w) * (SCREEN_WIDTH as f32)).round() as isize;
+                    let py_i = ((prev_y / h) * (SCREEN_HEIGHT as f32)).round() as isize;
```

Comment:
I see what you mean. I would love to have this kind of logic baked into the public API in some form. Admittedly it's just a feature I didn't think of at the time. But seems so obvious now.